pkg/reconciler/managed/reconciler.go

```diff
@@ -1106,6 +1121,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 	}
 
+	// record the drift after the successful update.
+	r.driftRecorder.recordUpdate(managed.GetName())
```

Comment:
Should we also reset the timer here (i.e. call `recordUnchanged`)? I believe we rely on an upcoming reconciliation which we expect to happen after a successful update, right?

Reply:
there would be an unchanged case afterwards. But I agree, we can directly reset.

Reply:
done